R/read_bed.r

```diff
@@ -26,9 +26,11 @@
 #' read_bed(valr_example('3fields.bed.gz'), sort = FALSE)
 #'
 #' @export
-read_bed <- function(filename, n_fields = 3, col_types = bed12_coltypes,
+read_bed <- function(filename, n_fields = NULL, col_types = bed12_coltypes,
                      sort = TRUE, ...) {
 
+  if (missing(n_fields)) n_fields <- ncol(read.table(filename, nrows = 5,sep="\t"))
```

Comment:
Will do.

Also, I'm currently determining the number of columns arbitrarily on the first 5 rows of the input file... any reason to change that? I suppose two rows should be sufficient in theory for a valid bed file (w/ or w/o a track header)?